imod_coupler/drivers/dfm_metamod/exchange_collector.py

```diff
@@ -0,0 +1,87 @@
+import abc
+import os
+from pathlib import Path
+from typing import Any
+
+import netCDF4 as nc
+import numpy as np
+from numpy.typing import NDArray
+
+
+class AbstractExchange(abc.ABC):
+    def __init__(self, name: str):
+        pass
+
+    def write_exchange(self, exchange: NDArray[Any], time: float) -> None:
+        pass
+
+
+class NetcdfExchangeLogger(AbstractExchange):
+    output_file: Path
+    name: str
+
+    def __init__(self, name: str, output_dir: Path, properties: dict[str, Any]):
+        if not (os.path.isdir(output_dir)):
+            os.mkdir(output_dir)
+        output_file = Path.joinpath(output_dir, name + ".nc")
+        self.ds = nc.Dataset(output_file, "w")
+        self.name = name
+
+    def initfile(self, ndx: int) -> None:
+        self.nodedim = self.ds.createDimension("id", ndx)
+        self.timedim = self.ds.createDimension("time",)
+        self.timevar = self.ds.createVariable("time", "f8", ("time",))
+        self.datavar = self.ds.createVariable("xchg", "f8", ("time", "id",))
+        self.pos = 0
+
+    def write_exchange(
+        self, exchange: NDArray[Any], time: float, sync: bool = False
+    ) -> None:
+        if len(self.ds.dimensions) == 0:
+            self.initfile(len(exchange))
+        loc = np.where(self.timevar[:] == time)
+        if np.size(loc) > 0:
+            self.datavar[loc[0], :] = exchange[:]
+        else:
+            self.timevar[self.pos] = time
```

Comment:
do we need additional check if time(i) > time(i-1)?

Reply:
it doesn't need to be. This can be seen in test test_exchange_collector_overwrites_when_time_is_repeated  where we 
add logging for time 8, then 9 , then 10, and then 9 again. 


    exchange_collector.log_exchange("example_stage_output", some_arrays[0], 8.0)
    exchange_collector.log_exchange("example_stage_output", some_arrays[1], 9.0)
    exchange_collector.log_exchange("example_stage_output", some_arrays[2], 10.0)
    exchange_collector.log_exchange("example_stage_output", some_arrays[3], 9.0)